src/explorer/SiteTreeItem.ts

```diff
@@ -212,21 +213,22 @@ export abstract class SiteTreeItem extends AzureParentTreeItem<ISiteTreeRoot> {
 
     }
 
-    private getIgnoredFoldersForDeployment(runtime: string): string[] {
+    private async getIgnoredFoldersForDeployment(fsPath: string, runtime: string): Promise<string[]> {
         let ignoredFolders: string[];
         switch (runtime) {
             case LinuxRuntimes.node:
                 ignoredFolders = ['node_modules{,/**}'];
                 break;
             case LinuxRuntimes.python:
+                const venvsfsPaths: string[] = (await venvUtils.getExistingVenvs(fsPath)).map(venvPath => `${venvPath}{,/**}`);
                 // list of Python ignorables are pulled from here https://github.com/github/gitignore/blob/master/Python.gitignore
                 // Byte-compiled / optimized / DLL files
                 ignoredFolders = ['__pycache__{,/**}', '*.py[cod]', '*$py.class',
                     // Distribution / packaging
                     '.Python{,/**}', 'build{,/**}', 'develop-eggs{,/**}', 'dist{,/**}', 'downloads{,/**}', 'eggs{,/**}', '.eggs{,/**}', 'lib{,/**}', 'lib64{,/**}', 'parts{,/**}', 'sdist{,/**}', 'var{,/**}',
-                    'wheels{,/**}', 'share/python-wheels{,/**}', '*.egg-info{,/**}', '.installed.cfg', '*.egg', 'MANIFEST',
-                    // Environments
-                    '.env{,/**}', '.venv{,/**}', 'env{,/**}', 'venv{,/**}', 'ENV{,/**}', 'env.bak{,/**}', 'venv.bak{,/**}'];
+                    'wheels{,/**}', 'share/python-wheels{,/**}', '*.egg-info{,/**}', '.installed.cfg', '*.egg', 'MANIFEST'];
+                // Virtual Environments
+                ignoredFolders = ignoredFolders.concat(venvsfsPaths);
```

Comment:
I would include the previous list of virtual environments that you had. It will help if you fail to detect a venv, if the venv doesn't exist yet, or if someone on a different machine (with the same repo) uses a different venv name